components/insight/SRC/org/openmicroscopy/shoola/env/data/views/calls/ThumbnailLoader.java

```diff
@@ -259,198 +266,78 @@ public void doCall() {
      * Returns the lastly retrieved thumbnail.
      * This will be packed by the framework into a feedback event and
      * sent to the provided call observer, if any.
-     * 
+     *
      * @return A {@link ThumbnailData} containing the thumbnail pixels.
      */
-    protected Object getPartialResult() { return currentThumbnail; }
+    protected Object getPartialResult() {
+        return currentThumbnail;
+    }
 
     /**
      * Returns the last loaded thumbnail (important for the BirdsEyeLoader to
-     * work correctly). But in fact, thumbnails are progressively delivered with 
-     * feedback events. 
+     * work correctly). But in fact, thumbnails are progressively delivered with
+     * feedback events.
+     *
      * @see BatchCallTree#getResult()
      */
-    protected Object getResult() { return currentThumbnail; }
+    @Override
+    protected Object getResult() {
+        return currentThumbnail;
+    }
 
     /**
      * Creates a new instance.
      * If bad arguments are passed, we throw a runtime exception so to fail
      * early and in the caller's thread.
-     * 
-     * @param ctx The security context.
-     * @param imgs Contains {@link DataObject}s, one 
-     *             for each thumbnail to retrieve.
-     * @param maxWidth The maximum acceptable width of the thumbnails.
+     *
+     * @param ctx       The security context.
+     * @param imgs      Contains {@link DataObject}s, one
+     *                  for each thumbnail to retrieve.
+     * @param maxWidth  The maximum acceptable width of the thumbnails.
      * @param maxHeight The maximum acceptable height of the thumbnails.
-     * @param userIDs The users the thumbnail are for.
+     * @param userIDs   The users the thumbnail are for.
      */
-    public ThumbnailLoader(SecurityContext ctx, Set<DataObject> imgs,
-            int maxWidth, int maxHeight, Set<Long> userIDs)
-    {
-        if (imgs == null) throw new NullPointerException("No images.");
-        if (maxWidth <= 0)
+    public ThumbnailLoader(SecurityContext ctx, Collection<DataObject> imgs,
+                           int maxWidth, int maxHeight, Collection<Long> userIDs) {
+        if (images == null) {
```

Comment:
That's the NPE... Typo, has to be `imgs`.

Reply:
Concur: ```
Abnormal termination due to an uncaught exception.
java.lang.NullPointerException: No images.
        at org.openmicroscopy.shoola.env.data.views.calls.ThumbnailLoader.<init>(ThumbnailLoader.java:303)
        at org.openmicroscopy.shoola.env.data.views.calls.ThumbnailLoader.<init>(ThumbnailLoader.java:330)
```